demos/python_demos/formula_recognition_demo/formula_recognition_demo.py

```diff
@@ -16,68 +16,129 @@
-    image_raw = cv.copyMakeBorder(image_raw, 0, target_height - img_h,
-                                  0, target_width - img_w, cv.BORDER_CONSTANT,
-                                  None, COLOR_WHITE)
-    image = image_raw.transpose((2, 0, 1))
-    return np.expand_dims(image, axis=0)
+from utils import *
+
+
+class InteractiveDemo:
+    def __init__(self, input_model_shape, resolution):
+        self.resolution = resolution
+        self._tgt_shape = input_model_shape
+        self.start_point, self.end_point = self._create_input_window()
+        self._prev_rendered_formula = None
+        self._prev_formula_img = None
+        self._latex_h = 0
+        self._renderer = create_renderer()
+
+    def _create_input_window(self):
+        aspect_ratio = self._tgt_shape[0] / self._tgt_shape[1]
+        width = min(DEFAULT_WIDTH, self.resolution[0])
+        height = int(width * aspect_ratio)
+        start_point = (int(self.resolution[0] / 2 - width / 2), int(self.resolution[1] / 2 - height / 2))
+        end_point = (int(self.resolution[0] / 2 + width / 2), int(self.resolution[1] / 2 + height / 2))
+        return start_point, end_point
+
+    def get_crop(self, frame):
+        crop = frame[self.start_point[1]:self.end_point[1], self.start_point[0]:self.end_point[0], :]
+        return crop
+
+    def _draw_rectangle(self, frame):
+        frame = cv.rectangle(frame, self.start_point, self.end_point, color=RED, thickness=2)
+        return frame
+
+    def resize_window(self, action):
+        height = self.end_point[1] - self.start_point[1]
+        width = self.end_point[0] - self.start_point[0]
+        aspect_ratio = height / width
+        if action == 'increase':
+            max_h, max_w = min(MAX_HEIGHT, self.resolution[1]), min(MAX_WIDTH, self.resolution[0])
+            if height >= max_h or width >= max_w:
+                return
+            self.start_point = (self.start_point[0] - DEFAULT_RESIZE_STEP,
+                                self.start_point[1] - int(DEFAULT_RESIZE_STEP * aspect_ratio))
+            self.end_point = (self.end_point[0] + DEFAULT_RESIZE_STEP,
+                              self.end_point[1] + int(DEFAULT_RESIZE_STEP * aspect_ratio))
+        elif action == 'decrease':
+            if height <= MIN_HEIGHT or width <= MIN_WIDTH:
+                return
+            self.start_point = (self.start_point[0] + DEFAULT_RESIZE_STEP,
+                                self.start_point[1] + int(DEFAULT_RESIZE_STEP * aspect_ratio))
+            self.end_point = (self.end_point[0] - DEFAULT_RESIZE_STEP,
+                              self.end_point[1] - int(DEFAULT_RESIZE_STEP * aspect_ratio))
+        else:
+            raise ValueError("wrong action: {}".format(action))
+
+    def _put_text(self, frame, text):
+        if text == '':
+            return frame
+        text = strip_internal_spaces(text)
+        (txt_w, self._latex_h), baseLine = cv.getTextSize(text, cv.FONT_HERSHEY_SIMPLEX, 1, 3)
```

Comment:
The `baseLine` should be renamed to `base_line`. We use `snake_case` for variables. But also it should be removed, since it is unused :)

Reply:
Removed